packages/apps/graph/src/graph/objects/Block.ts

```diff
@@ -66,3 +73,25 @@ async function getTransactionsRequestkeyByEvent(
       AND t.block = ${parent.hash}`
   ).map((r) => r.requestkey);
 }
+
+async function getConfirmationDepth(blockHash: string): Promise<number> {
+  const result = await prismaClient.$queryRaw<{ depth: number }[]>`
+    WITH RECURSIVE BlockDescendants AS (
+      SELECT hash, parent, 0 AS depth
+      FROM blocks
+      WHERE hash = ${blockHash}
+      UNION ALL
+      SELECT b.hash, b.parent, d.depth + 1 AS depth
+      FROM BlockDescendants d
+      JOIN blocks b ON d.hash = b.parent
+    )
+    SELECT MAX(depth) AS depth
+    FROM BlockDescendants;
+  `;
+
+  if (result && result[0] && result[0].depth) {
```

Comment:
`result[0]` would throw an error if there isn't anything in the array, and `result` is always true. Consider doing `result.length && result[0].depth`